### DISCUSSION TITLE
Collapsible sidebar, preferably with a keyboard shortcut

### DISCUSSION THREAD

**Discussion by hirehamir** [withastro/starlight#2561] [Feature Requests]
### What version of `starlight` are you using?

latest

### What is your idea?

It'd be nice if users could collapse the sidebar.

Ideally, there should be a keyboard shortcut to toggle the sidebar's state. A common one that developers may be familiar with is Ctrl + B.

If there is a collapse/expand sidebar button, the button should show this keyboard shortcut on hover.

### Why is this feature necessary?

The sidebar takes up a non-trivial amount of space in portrait mode.

### Do you have examples of this feature in other projects?

https://rust-lang.github.io/mdBook/ lets users collapse the sidebar by clicking on the three horizontal lines, although there's no keyboard shortcut.

Also, the button to collapse the sidebar lives outside the sidebar, which is a bit unexpected.

The button to collapse the sidebar should be in the sidebar.

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by hirehamir** [comment]
![image](https://github.com/user-attachments/assets/ca32a7b5-610e-40a0-91ee-19ca8d9367ce)

**Comment by HiDeoo** [comment]
Thanks for the feedback 🙌 

This is a duplicate of https://github.com/withastro/starlight/discussions/2264 so I'll close this one so we can keep the discussion and gather feedback in one place.

**Reply by hirehamir** [reply]
You're welcome! We can do that.
